datafusion/core/src/datasource/physical_plan/parquet.rs

```diff
@@ -148,7 +149,7 @@ mod tests {
 
             let mut source = ParquetSource::default();
             if let Some(predicate) = predicate {
-                source = source.with_predicate(Arc::clone(&file_schema), predicate);
+                source = source.with_predicate(predicate);
```

Comment:
This seemed like an easy win since I was able to just change this so that the schema is _always_ passed in by the `FileSourceConfigBuilder` instead of only when `with_predicate` is called.
This was necessary becasue `with_predicate` is no longer called to attach a predicate, instaed it happens during an optimization pass so `ParquetSource` neesd to have it available at that point.
I left `with_predicate` in there to avoid churn and in case there is a use case for attaching a predicate directly through the scan instad of a as a FilterExec that later gets pushed into the scan.